plaso/storage/sqlite/sqlite_file.py

```diff
@@ -646,6 +687,24 @@ def AddEvent(self, event, serialized_data=None):
     self._UpdateEventDataIdentifierBeforeSerialize(event)
     self._AddSerializedEvent(event)
 
+  def AddEventData(self, event_data, serialized_data=None):
+    """Adds event data.
+
+    Args:
+      event_data (EventData): event data.
+      serialized_data (Optional[bytes]): serialized form of the event data.
+
+    Raises:
+      IOError: when the storage file is closed or read-only.
+      OSError: when the storage file is closed or read-only.
+    """
+    self._RaiseIfNotWritable()
+
+    # The serialized data is not used, as this method modifies the attribute
```

Comment:
Comment: this to me suggests that attaching the EventDataStream to the event vs. the EventData is a better way to approach this. Events are cheap to serialize and deserialize, as they have a relatively small number of fixed fields. EventData are typically much larger, and have a variable number of fields.

Having to reserialize EventData when merging to the session store negates much of the benefit of doing any serialized data caching at all.

Reply:
Ack, let's chat more about this. There are more events than event data so the costs/benefits might even out. I opt to continue with this path, we can definitely look into gains of changing this at a later date. Especially if we're considering serializing to native SQLite types,